### PR TITLE
feat: Tab Switcher View

### PR DESCRIPTION
**Pull request by whizzzkid**
Closes: #99 

In this PR:
- Introduced a new tabbed view based on: https://chakra-ui.net/docs/components/tabs/usage
- renamed `useViewMode` -> `getViewMode`
- cleaned up more interfaces and apis wherever I could.

Q: @juliaxbow @SgtPooki at some point "?view=detail" functionality was lost, is that a requirement? e.g.: https://www.starmaps.app/roadmap/github.com/ipfs/roadmap/issues/96?view=detail won't take you to detailed view by default, instead would render the view that last set by `setViewMode`

### CODE REVIEW

**Review by juliaxbow — COMMENTED** [review]
Tabbed view looks good to me! Will let @SgtPooki review the code

**Comment by SgtPooki** [comment]
The viewMode was moved to be clientside only so we werent doing a full page refresh for data we already had, and so the action was faster.

Also, hooks should be named 'use<name>'

File: hooks/useViewMode.ts [hunk 1]
```diff
@@ -59,6 +59,6 @@ const customStateFunction: typeof useState = <S = typeof ViewMode>(initialState?
   return [state, setCachedState] as [S | undefined, Dispatch<SetStateAction<S | undefined>>];
 }
 
-const [useViewMode, setViewMode] = useSharedHook(customStateFunction, ViewMode.Simple);
+const [getViewMode, setViewMode] = useSharedHook(customStateFunction, ViewMode.Simple);
 
-export { useViewMode, setViewMode };
+export { getViewMode, setViewMode };
```
Comment (SgtPooki) [inline hooks/useViewMode.ts:64]: we should stick to naming conventions of hooks.

`getViewMode` could be mistaken for a non-hook.

File: components/roadmap-grid/group-item.tsx [hunk 2]
```diff
@@ -11,13 +11,11 @@ import styles from './Roadmap.module.css';
  * This is the component for the Group header (themes)
  * @returns {JSX.Element}
  */
-export function GroupItem({issueData, group }: GroupItemProps) {
-  const viewMode = useViewMode();
-  let detailedViewClass = '';
+export function GroupItem({ group }: GroupItemProps) {
+  const viewMode = getViewMode();
```
Comment (SgtPooki) [inline components/roadmap-grid/group-item.tsx:15]: `useViewMode`

File: components/roadmap-grid/group-wrapper.tsx [hunk 3]
```diff
@@ -1,9 +1,9 @@
-import { useViewMode } from '../../hooks/useViewMode';
+import { getViewMode } from '../../hooks/useViewMode';
 import { ViewMode } from '../../lib/enums';
 import styles from './Roadmap.module.css';
 
 export function GroupWrapper({ children, cssName = '' }) {
-  const viewMode = useViewMode();
+  const viewMode = getViewMode();
```
Comment (SgtPooki) [inline components/roadmap-grid/group-wrapper.tsx:6]: `useViewMode`

File: lib/types.d.ts [hunk 4]
```diff
@@ -114,7 +114,6 @@ export interface GroupItemProps {
    * The root node issue data; The GitHub issue URL provided in RoadmapForm, or
```
Comment (SgtPooki) [inline lib/types.d.ts:114]: it looks like this comment can go away now

**Review by SgtPooki — CHANGES_REQUESTED** [review]

**Review by SgtPooki — APPROVED** [review]

**Pull request merged**